src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -328,31 +317,169 @@ public void Validate(FixedSizeSchemaIndexDef actual)
+            }
+
+            public override ByteStringContext.Scope GetSlice(ByteStringContext context, TableValueBuilder value,
+                out Slice slice)
+            {
+                using (context.Allocate(value.Size, out var buffer))
+                {
+                    value.CopyTo(buffer.Ptr);
+                    var reader = value.CreateReader(buffer.Ptr);
+                    return IndexValueGenerator(context, ref reader, out slice);
+                }
+            }
+
+            public override byte[] Serialize()
+            {
+                // We serialize the Type enum as ulong to be "future-proof"
+                var castedType = (long)Type;
+
+                var serializer = new TableValueBuilder
+                {
+                    castedType,
+                    IsGlobal,
+                    Name
+                };
+
+                var methodNameBytes = Encodings.Utf8.GetBytes(IndexValueGenerator.Method.Name);
+                fixed (byte* ptr = methodNameBytes)
+                {
+                    serializer.Add(ptr, methodNameBytes.Length);
+                }
+
+                Debug.Assert(IndexValueGenerator.Method.DeclaringType?.AssemblyQualifiedName != null, 
+                    $"Invalid {nameof(IndexValueGenerator)} '{IndexValueGenerator.Method.Name}'");
+
+                var declaringTypeBytes = Encodings.Utf8.GetBytes(IndexValueGenerator.Method.DeclaringType.AssemblyQualifiedName);
+                fixed (byte* ptr = declaringTypeBytes)
+                {
+                    serializer.Add(ptr, declaringTypeBytes.Length);
+                }
+
+                byte[] serialized = new byte[serializer.Size];
+
+                fixed (byte* destination = serialized)
+                {
+                    serializer.CopyTo(destination);
+                }
+
+                return serialized;
+            }
+
+            public static DynamicBTreeIndexDef ReadFrom(ByteStringContext context, byte* location, int size)
+            {
+                var input = new TableValueReader(location, size);
+                var indexDef = new DynamicBTreeIndexDef();
+
+                byte* currentPtr = input.Read(1, out _);
+                indexDef.IsGlobal = Convert.ToBoolean(*currentPtr);
+
+                currentPtr = input.Read(2, out var currentSize);
+                Slice.From(context, currentPtr, currentSize, ByteStringType.Immutable, out indexDef.Name);
+
+                // read IndexValueGenerator method name
+                currentPtr = input.Read(3, out currentSize);
+                var methodName = Encodings.Utf8.GetString(currentPtr, currentSize);
+
+                // read IndexValueGenerator declaring type
+                currentPtr = input.Read(4, out currentSize);
+                var declaringType = Encodings.Utf8.GetString(currentPtr, currentSize);
+
+                var type = System.Type.GetType(declaringType);
```

Comment:
For security reasons, we probably want to do that only on types that has a well known attribute that is meant to be used for that.
Otherwise, we may have a code execution on untrusted binaries.

Reply:
done